golem-cli/src/command_handler/api/deployment.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::command::api::deployment::ApiDeploymentSubcommand;
+use crate::command::shared_args::ProjectNameOptionalArg;
+use crate::command_handler::Handlers;
+use crate::context::{Context, GolemClients};
+use crate::error::service::AnyhowMapServiceError;
+use crate::error::NonSuccessfulExit;
+use crate::model::text::fmt::log_error;
+use crate::model::{ApiDefinitionId, ApiDefinitionIdWithVersion, ApiDeployment};
+use anyhow::bail;
+use golem_client::api::ApiDeploymentClient as ApiDeploymentClientOss;
+use golem_client::model::{
+    ApiDefinitionInfo as ApiDefinitionInfoOss, ApiDeploymentRequest as ApiDeploymentRequestOss,
+    ApiSite as ApiSiteOss,
+};
+use golem_cloud_client::api::ApiDeploymentClient as ApiDeploymentClientCloud;
+use golem_cloud_client::model::{
+    ApiDefinitionInfo as ApiDefinitionInfoCloud, ApiDeploymentRequest as ApiDeploymentRequestCloud,
+    ApiSite as ApiSiteCloud,
+};
+use golem_wasm_rpc_stubgen::log::{log_warn_action, LogColorize};
+use std::sync::Arc;
+
+pub struct ApiDeploymentCommandHandler {
+    ctx: Arc<Context>,
+}
+
+impl ApiDeploymentCommandHandler {
+    pub fn new(ctx: Arc<Context>) -> Self {
+        Self { ctx }
+    }
+
+    pub async fn handle_command(&mut self, command: ApiDeploymentSubcommand) -> anyhow::Result<()> {
+        match command {
+            ApiDeploymentSubcommand::Deploy {
+                project,
+                definitions,
+                host,
+                subdomain,
+            } => self.deploy(project, definitions, host, subdomain).await,
+            ApiDeploymentSubcommand::Get { site } => self.get(site).await,
+            ApiDeploymentSubcommand::List {
+                project,
+                definition,
+            } => self.list(project, definition).await,
+            ApiDeploymentSubcommand::Delete { site } => self.delete(site).await,
+        }
+    }
+
+    async fn deploy(
+        &self,
+        project: ProjectNameOptionalArg,
+        api_defs: Vec<ApiDefinitionIdWithVersion>,
+        host: String,
+        subdomain: Option<String>,
+    ) -> anyhow::Result<()> {
+        let project = self
```

Comment:
should this be moved into the cloud branch of the match?

Reply:
no